src/ConnectedMode/Binding/ImportBeforeFileGenerator.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using System.Text;
+using System.Xml;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    internal class ImportBeforeFileGenerator
+    {
+        private readonly IFileSystem fileSystem;
+
+        private const string targetsFileName = "SonarLint.targets";
+
+        public ImportBeforeFileGenerator() : this(new FileSystem()) { }
+
+        public ImportBeforeFileGenerator(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+
+            CreateTargetsFileIfNotExists();
+        }
+
+        private void CreateTargetsFileIfNotExists()
+        {
+            var sb = new StringBuilder();
+            var xmlWriter = Create(sb);
+
+            xmlWriter.WriteStartElement("Project", "http://schemas.microsoft.com/developer/msbuild/2003");
+            xmlWriter.WriteAttributeString("xmlns", "http://schemas.microsoft.com/developer/msbuild/2003");
+
```

Comment:
I already have the code to ad din the globalAnalyzers however, I dont actually know the paths of said files here so I wasn't sure it made sense to include it. 

```
private void WriteGlobalAnalyzerConfigFile(XmlWriter xmlWriter, string condition, string pathToConfig)
        {
            xmlWriter.WriteStartElement("GlobalAnalyzerConfigFiles");
            xmlWriter.WriteAttributeString("Condition", $"'$(Language)' == '{condition}'");
            xmlWriter.WriteAttributeString("Include", pathToConfig);
            xmlWriter.WriteEndElement();
        }
```

Reply:
Building the file content using an XmlWriter is going to be hard to read and maintain. Because the content of the file is almost completely static, we could take a simpler approach and just use a hard-coded string.
If we do need to substitute data into the file can just put a known placeholder in the text and using `String.Replace` e.g.

e.g. 

```

string content = @"<Project...>

  <ItemGroup Condition="" $(RunningInsideVisualStudio) == 'true' "">
       <SLVSConfig>[[PATH_PLACEHOLDER]]</SLVSConfig>
  </ItemGroup>

</Project>"

content = content.Replace("[[PATH_PLACEHOLDER]]", filePath);
```

We can take this a step further; instead of defining the content as a string in code, we can embed a text file in the assembly, and then just read the contents of the file at runtime using `Assembly.GetManifestResourceStream`.

That way, when we want to change the content, we just need to edit the text file.
To embed a text file, we just need to add a text file to the project and set the `Build Action` to `Embedded Resource`.

We do this in a couple of places in the product already; for example, the rule help files are all embedded as text files.
Let's have a chat about this offline after the sync.

Reply:
e.g. from SimpleRuleHelpBuilderTests.cs

```
        private static string ReadResource(string fullResourceName)
        {
            using var stream = new StreamReader(ResourceAssembly.GetManifestResourceStream(fullResourceName));
            return stream.ReadToEnd();
        }

```